include/openbabel/generic.h

```diff
@@ -1185,9 +1185,71 @@ namespace OpenBabel
 
   };
 
- //! A standard iterator over vectors of OBGenericData (e.g., inherited from OBBase)
-  typedef std::vector<OBGenericData*>::iterator OBDataIterator;
+  class OBAPI OBPcharge
+  {
+  protected:
+    int         _atomIndex;
+    double      _q;
+  public:
+    OBPcharge() {};
+    OBPcharge(int atomIndex, double q) {_atomIndex = atomIndex; _q = q ;}
+    ~OBPcharge() {};
+
+    double GetQ() {return _q;}
+    int    GetAtomIndex() {return _atomIndex;}
+    void   SetQ(double q) {_q = q;}
+    void   SetAtomIndex(int atomIndex) {_atomIndex = atomIndex;}
+    
+  };
+
+  typedef std::vector<OBPcharge*>::iterator OBPchargeIterator;
+  
+  class OBAPI OBPcharges: public OBGenericData
```

Comment:
This makes little sense. I'm open to storing multiple sets of named partial charges from a QM file. No problem.

But this is hugely inefficient. For one, there's already support for an atomic partial charge OBAtom::SetPartialCharge() - which doesn't require extra storage.

Even if you want to have this class as a generic data type, the storage here is inefficient. Just keep a vector<double> for the charges? Why bother with the atom index business? Just index into the vector!